### ISSUE TITLE
Error when leveldb is not available

### ISSUE THREAD

**Issue by devos50** [Tribler/tribler#3200]
Received this one in the error reporter:

```
Traceback (most recent call last):
File "TriblerGUI\event_request_manager.py", line 99, in on_read_data
RuntimeError: Unhandled Error
Traceback (most recent call last):
File "lib\multiprocessing\process.py", line 114, in run

File "TriblerGUI\core_manager.py", line 63, in start_tribler_core

File "lib\site-packages\twisted\internet\base.py", line 1243, in run

File "lib\site-packages\twisted\internet\base.py", line 1252, in mainLoop

--- <exception caught here> ---
File "lib\site-packages\twisted\internet\base.py", line 851, in runUntilCurrent

File "Tribler\Core\TFTP\handler.py", line 233, in data_came_in

File "Tribler\Core\TFTP\handler.py", line 301, in _handle_new_request

File "Tribler\Core\TFTP\handler.py", line 345, in _load_torrent

File "lib\_abcoll.py", line 382, in get

File "Tribler\Core\leveldbstore.py", line 86, in __getitem__

exceptions.AttributeError: 'LevelDbStore' object has no attribute '_db'
```

**Comment by qstokkink** [comment]
This was fixed on `devel` in `8ac32d542c0cf51942b74a50af24a3be4820f354`, we will need to backport these fixes to `next`.

**Comment by devos50** [comment]
This issue is not fixed unfortunately, see the following crash report:

```
Traceback (most recent call last):
File "TriblerGUI\event_request_manager.py", line 99, in on_read_data
RuntimeError: Unhandled Error
Traceback (most recent call last):
File "lib\multiprocessing\process.py", line 114, in run

File "TriblerGUI\core_manager.py", line 65, in start_tribler_core

File "lib\site-packages\twisted\internet\base.py", line 1243, in run

File "lib\site-packages\twisted\internet\base.py", line 1252, in mainLoop

--- <exception caught here> ---
File "lib\site-packages\twisted\internet\base.py", line 851, in runUntilCurrent

File "Tribler\Core\TFTP\handler.py", line 234, in data_came_in

File "Tribler\Core\TFTP\handler.py", line 306, in _handle_new_request

File "Tribler\Core\TFTP\handler.py", line 350, in _load_torrent

File "lib\_abcoll.py", line 382, in get

File "Tribler\Core\leveldbstore.py", line 86, in __getitem__

exceptions.AttributeError: 'LevelDbStore' object has no attribute '_db'
```

**Comment by devos50** [comment]
I received a detailed report from the same user apparently:

```
leveldb.LevelDBError: Corruption: 2 missing files; e.g.: ..\..\Users\<SECRET_USER>\AppData\Roaming\.Tribler\collected_torrents/000055.sst
```

**Comment by devos50** [comment]
I'm not sure whether we can fix this error before the Tribler 7 release without an RC5. I propose to postpone this issue to 7.1. @qstokkink @egbertbouman @xoriole what do you think?

**Comment by qstokkink** [comment]
Can't we just add a check to `data_came_in` which checks for `self.session.config.get_metadata_enabled()` and `self.session.lm.torrent_store is not None`?

**Comment by devos50** [comment]
@qstokkink unfortunately, that's not a solution for this problem. If the leveldb store cannot be created, the `_db` variable in `LevelDbStore` will be `None`, even if the metadata/torrent store is enabled.

**Comment by qstokkink** [comment]
Woop sorry, I meant: `self.session.lm.torrent_store._db is not None`

**Comment by devos50** [comment]
That would work, but I consider it as a workaround for the underlying problem. We have to get the exact report (which is raised now after #3285) before we are able to accurately debug this issue. This in turns requires either a 5th release candidate or investigating it after the official release of Tribler 7. Depending on the number of outstanding issues, I'm not sure whether we need an additional release candidate but we still have to discuss that.

**Comment by qstokkink** [comment]
I believe this can now no longer happens due to this change:
https://github.com/Tribler/tribler/blob/4124b4f39c1e230b3023828ed1aff6fdc43e23dc/Tribler/Core/leveldbstore.py#L77-L78

Tribler crashes when the LevelDbStore._db is None, so it can never be None in other classes.

**Comment by devos50** [comment]
True but that fix is only there to better debug the code (find what’s actually causing the error) so we should wait before a new, more descriptive exception pops up.

As much as I want this error to be gone, we should consider all kind of errors when creating the leveldb store and handle them accordingly.

**Comment by devos50** [comment]
We haven't observed a more detailed report so far. I will postpone this issue.

**Comment by devos50** [comment]
Closing this issue due to lack of feedback.
